Components/MineSharp.Protocol/Packets/Clientbound/Play/AwardStatisticsPacket.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System.Collections.Frozen;
+using MineSharp.Core.Serialization;
+using MineSharp.Data;
+using MineSharp.Data.Protocol;
+
+namespace MineSharp.Protocol.Packets.Clientbound.Play;
+
+/// <summary>
+///     Award statistics packet sent as a response to Client Command.
+/// </summary>
+/// <param name="Count">Number of elements in the statistics array.</param>
+/// <param name="Statistics">Array of statistics.</param>
+public sealed record AwardStatisticsPacket(int Count, AwardStatisticsPacket.Statistic[] Statistics) : IPacket
+{
+    /// <inheritdoc />
+    public PacketType Type => StaticType;
+    /// <inheritdoc />
+    public static PacketType StaticType => PacketType.CB_Play_Statistics;
+
+    /// <inheritdoc />
+    public void Write(PacketBuffer buffer, MinecraftData version)
+    {
+        buffer.WriteVarInt(Count);
+        foreach (var statistic in Statistics)
+        {
+            buffer.WriteVarInt((int)statistic.CategoryId);
+            buffer.WriteVarInt((int)statistic.StatisticId);
+            buffer.WriteVarInt(statistic.Value);
+        }
+    }
+
+    /// <inheritdoc />
+    public static IPacket Read(PacketBuffer buffer, MinecraftData version)
+    {
+        var count = buffer.ReadVarInt();
+        var statistics = new Statistic[count];
+        for (int i = 0; i < count; i++)
+        {
+            var categoryId = (CategoryId)buffer.ReadVarInt();
+            var statisticId = (StatisticId)buffer.ReadVarInt();
+            var value = buffer.ReadVarInt();
+            statistics[i] = new Statistic(categoryId, statisticId, value);
+        }
+
+        return new AwardStatisticsPacket(count, statistics);
+    }
+
+    /// <summary>
+    ///     Represents a single statistic entry.
+    /// </summary>
+    /// <param name="CategoryId">The category ID of the statistic.</param>
+    /// <param name="StatisticId">The statistic ID.</param>
+    /// <param name="Value">The value of the statistic.</param>
+    public sealed record Statistic(CategoryId CategoryId, StatisticId StatisticId, int Value);
+
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+    /// <summary>
+    ///     Enum representing different types of statistic IDs.
+    /// </summary>
```

Comment:
The numerical values of statistics come from a registry (minecraft:custom_stat) and can change with a new minecraft version (#73)

Reply:
Oh ok. That makes sense. wiki.vg did not make that clear. I leave it for now and change it once the Registry is implemented